# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/chrome/common/child_thread.cc
# Commit: 790fd8710dd7
# Full Hash: 790fd8710dd7bfa6085379e7f90b043e7aff383f
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 12a: Initialize NodeController when creating IO thread, r=handyman
#   
#   This also consumes the existing channel created when launching a process to
#   create the the conneciton required by NodeController for communicating between
#   processes. In part 12b, consumers of the broken APIs will be adjusted to use
# ==============================================================================

diff -r 7b7819250f79 -r 790fd8710dd7 ipc/chromium/src/chrome/common/child_thread.cc
--- a/ipc/chromium/src/chrome/common/child_thread.cc	Tue Jun 22 18:17:22 2021 +0000
+++ b/ipc/chromium/src/chrome/common/child_thread.cc	Tue Jun 22 18:17:22 2021 +0000
@@ -7,6 +7,7 @@
 #include "chrome/common/child_thread.h"
 
 #include "chrome/common/child_process.h"
+#include "mozilla/ipc/NodeController.h"
 
 ChildThread::ChildThread(Thread::Options options)
     : Thread("IPC I/O Child"),
@@ -35,12 +36,11 @@
 }
 
 void ChildThread::Init() {
-  channel_ = mozilla::MakeUnique<IPC::Channel>(channel_name_,
-                                               IPC::Channel::MODE_CLIENT, this);
+  auto channel = mozilla::MakeUnique<IPC::Channel>(
+      channel_name_, IPC::Channel::MODE_CLIENT, this);
+
+  initial_port_ =
+      mozilla::ipc::NodeController::InitChildProcess(std::move(channel));
 }
 
-void ChildThread::CleanUp() {
-  // Need to destruct the SyncChannel to the browser before we go away because
-  // it caches a pointer to this thread.
-  channel_ = nullptr;
-}
+void ChildThread::CleanUp() { mozilla::ipc::NodeController::CleanUp(); }